SYNTAX.md

```diff
@@ -30,13 +30,27 @@ These are sections that are specified directly within the `pipeline` argument cl
         * *Type*: bareword
         * *Description*: If given, node/image management will need to be specified explicitly in stages and no 
         automatic `checkout scm` call will occur.
-* *Takes a Closure*: No
+* *Takes a Closure*: yes
 * *Examples*:
-    * `agent label:'has-docker', docker:'ubuntu:lts'`
-    * `agent docker:'ubuntu:lts'`
-    * `agent label:'hi-speed'`
-    * `agent none`
-    * `agent any`
+
+```groovy
+agent {
+    label 'has-docker'
+    docker 'ubuntu:lts'
+}
+
+agent {
+    docker 'ubuntu:lts'
+}
+ 
+agent {
```

Comment:
So take a look at the discussion on the JIRA as well - but a couple main points.

One - `agent any { label "foo" }` isn't something I'm going to go with in the first place. I *really* don't like the look of `something somethingelse { closure }`. And...well, yeah, everything else is in the JIRA. I don't *love* this approach, but it works without a drastic rewrite of the backend implementation.